src/AudioIO.cpp

```diff
@@ -1748,12 +1774,28 @@ AudioThread::ExitCode AudioThread::Entry()
          gAudioIO->TrackBufferExchange();
          gAudioIO->mAudioThreadShouldCallTrackBufferExchangeOnce
             .store(false, std::memory_order_relaxed);
+
+         lastCommand = Command::eOnce;
       }
       else if( gAudioIO->mAudioThreadTrackBufferExchangeLoopRunning
          .load(std::memory_order_relaxed))
       {
          gAudioIO->TrackBufferExchange();
+
+         lastCommand = Command::eLoopRunning;
       }
+      else
+      {
+         if (lastCommand == Command::eLoopRunning)
+         {
+            // Main thread has told us to stop; (actually: to neither process "once" nor "loop running")
+            // acknoweldge that we received the order, and no more processing will be done.
+            gAudioIO->mAudioThreadStopAcknowledged.store(true, std::memory_order::memory_order_relaxed);
```

Comment:
Do not change treatment of `mAudioThreadTrackBufferExchangeLoopActive`.  I think that variable will be eliminated.  It is used only in the seeking code, which needs a serious rewriting.

Reply:
You detect the transition fom running to not-running and send acknowledgment.

The main thread waits for the acknowledgment when stopping the stream.

But recall what we discussed.  This isn't enough.  The main thread must make sure that this thread has first seen the transition from not-running to running, which should be done with another wait when starting the stream, and another acknowledgment sent from this stream.  (Maybe just one atomic that is an enumeration, not bool, is enough for this thread's acknowledgments.)

Without this other wait to acknowledge start, it is possible that the main thread could send a stop message soon after a start message but wait forever, because the worker never saw the transition to running, therefore never recognizes the transition back to not-running.

Maybe this is very unlikely for the normal stop, but maybe not so unlikely, for the other place where `mAudioThreadTrackBufferExchangeLoopRunning` becomes false, in the failure path for starting the stream.